common/sgx/eeid.c

```diff
@@ -390,6 +388,11 @@ static oe_result_t verify_base_image_signature(const sgx_sigstruct_t* sigstruct)
     oe_result_t result = OE_UNEXPECTED;
     unsigned char buf[sizeof(sgx_sigstruct_t)];
     size_t n = 0;
+#ifndef OE_BUILD_ENCLAVE
```

Comment:
Consider splitting this function into dedicated subfunctions for mbedtls and OpenSSL handling; the interleaved `#ifdefs` make the code increasingly hard to follow.

Side note: it looks like the mbedtls section of the code should have the same compiler issue, but may not be flagged because it is not being built with GCC in your current environment?

Reply:
This fixes both mbedTLS and OpenSSL, but  yeah, I think the time has come to split that up :-)

Reply:
(done)